test/common_nn.py

```diff
@@ -564,21 +564,21 @@ def _zero_grad_input(self, input):
 
     def _analytical_jacobian(self, module, input, jacobian_input=True, jacobian_parameters=True):
         output = self._forward(module, input)
+        output_size = output.nelement()
         output_t = output.data if isinstance(output, Variable) else output
-        d_out = output_t.new().resize_(output_t.size())
-        flat_d_out = d_out.view(-1)
 
         if jacobian_input:
-            jacobian_inp = self._jacobian(input, d_out.nelement())
+            jacobian_inp = self._jacobian(input, output_size)
             flat_jacobian_input = list(iter_tensors(jacobian_inp))
 
         if jacobian_parameters:
-            param, d_param = self._get_parameters(module)
-            num_param = sum(p.numel() for p in param)
-            jacobian_param = torch.zeros(num_param, d_out.nelement())
+            num_param = sum(p.numel() for p in self._get_parameters(module)[0])
+            jacobian_param = torch.zeros(num_param, output_size)
 
-        for i in range(flat_d_out.nelement()):
-            d_out.zero_()
+        for i in range(output_size):
+            _, d_param = self._get_parameters(module)
+            d_out = torch.zeros_like(output_t)
```

Comment:
I think you can still reuse `d_out`. It's not grad of any param; it's a tensor you created. The important part is reassigning `d_param` at every iteration

Reply:
Yes, this is stylistic: I think it's cleaner to re-create temporaries rather than calling `zero_()` every iteration.